Adding an obs space YAML for ATMS N21

keenaneure opened the pull request:
An observation space YAML is added for atms_n21 observtions for the 00z 27 May 2024 case.

An offline domain check, as well as bias correction, [following this previous issue](https://github.com/NOAA-EMC/RDASApp/issues/52), has been conducted.

SamuelDegelia-NOAA commented:
That is a weird failure. It looks like the automated testing did not clone all of the submodules correctly? Many of the directories in `/scratch1/NCEPDEV/fv3-cam/rrfsbot/PRs_RDASApp/376/sorc` are empty. This led to the build crashing because the `sed` replacement in `build.sh` cannot find `sorc/mpas/src/core_init_atmosphere/mpas_init_atm_cases.F`. Any idea what could cause that @guoqing-noaa? 

`/scratch1/NCEPDEV/fv3-cam` on Hera is only 89% full so I would not think that this is a space issue.

guoqing-noaa commented:
@SamuelDegelia-NOAA 
It was because JCSDA has reached their Git LFS limit

SamuelDegelia-NOAA commented:
Oops, the Hercules failure is because the obs were not synced to those other machines yet. That reminds me though, @keenaneure could you add `missing file action: "warn"` to the satellite radiance yamls? You can see the example for the other yamls in `rrfs-test/validated_yamls/templates/obtype_configs`.

SamuelDegelia-NOAA commented:
> @SamuelDegelia-NOAA It was because JCSDA has reached their Git LFS limit

Agh, not again. It seems to be resolved now though if the Hercules tests ran.

guoqing-noaa commented:
> > @SamuelDegelia-NOAA It was because JCSDA has reached their Git LFS limit
> 
> Agh, not again. It seems to be resolved now though if the Hercules tests ran.

I think the GIT LFS renew fee has been paid,  :)

keenaneure commented:
My apologies as I was in a meeting. I pushed those changes just now.

-Keenan

On Fri, Apr 25, 2025 at 2:06 PM rrfsbot ***@***.***> wrote:

> *rrfsbot* left a comment (NOAA-EMC/RDASApp#376)
> <https://github.com/NOAA-EMC/RDASApp/pull/376#issuecomment-2831335768>
>
> FAILED on hera
>
> started build_and_test on hera at UTC time: Fri Apr 25 20:06:02 UTC 2025
> finished at UTC time: Fri Apr 25 20:06:08 UTC 2025
>
> workdir: /scratch1/NCEPDEV/fv3-cam/rrfsbot/PRs_RDASApp/376
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/RDASApp/pull/376#issuecomment-2831335768>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG6NIPDIKJH24QMWD3L3MF323KIUPAVCNFSM6AAAAAB34BCS5OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMZRGMZTKNZWHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Keenan Eure, Ph.D.
Research Scientist
Cooperative Institute for Research in the Atmosphere (CIRA)
NOAA/OAR/Global Systems Laboratory, DSRC Room 2B415
325 Broadway, Boulder, CO 80305

SamuelDegelia-NOAA commented:
The yaml file looks good to me. @keenaneure would you mind addressing the linter error? 

`Error: 484:1 [trailing-spaces] trailing spaces`

I think that just means that there is some trailing whitespace at L484 for the yaml file.

xyzemc reviewed (approved):
Looks good to me.

guoqing-noaa commented on rrfs-test/validated_yamls/templates/obtype_config/atms_n21.yaml:
```diff
@@ -0,0 +1,585 @@
+     - obs space:
+         name: atms_n21
+         distribution:
```
@keenaneure Thanks a lot for adding atms_n21!

Could you add the common anchor variables to be consistent with other satellite YAML files?

Here is an instruction about the common anchor variables:
https://github.com/guoqing-noaa/satinfo2yaml/blob/main/README.md

Thanks!

guoqing-noaa replied:
FYI, this PR https://github.com/NOAA-EMC/RDASApp/pull/357 can also be referenced. Thanks!

guoqing-noaa replied:
@keenaneure  Thanks a lot for adding this!

keenaneure commented:
Thanks all for the feedback. I removed the trailing white space on L484, added anchor variables, and changed the use_flag sections for step 11 and step 12 filters to reflect other ATMS YAMLs. Pushing these changes now.

SamuelDegelia-NOAA commented:
I see that the new anchor `&atms_n21_obserr_bound_max` is not actually used in the yaml after it is defined. It is the same for `atms_n20.yaml` with `&atms_n20_obserr_bound_max`. But a similar variable is used in `abi_g16.yaml` and `abi_g18.yaml` for a final gross check filter. Is that filter not needed for ATMS data? @xyzemc

xyzemc commented:
> I see that the new anchor `&atms_n21_obserr_bound_max` is not actually used in the yaml after it is defined. It is the same for `atms_n20.yaml` with `&atms_n20_obserr_bound_max`. But a similar variable is used in `abi_g16.yaml` and `abi_g18.yaml` for a final gross check filter. Is that filter not needed for ATMS data? @xyzemc
Thanks @SamuelDegelia-NOAA  to point this out. I will check why atms and abi doing the gross check differently.

guoqing-noaa commented on rrfs-test/validated_yamls/templates/obtype_config/atms_n21.yaml:
```diff
@@ -0,0 +1,590 @@
+             obsfile: jdiag_atms_n21.nc4
+         simulated variables: [brightnessTemperature]
+         observed variables: [brightnessTemperature]
+         channels: &atms_n21_channels 1-22
```
Line 30:  change to 
```
         channels: *atms_n21_channels
```

guoqing-noaa commented on rrfs-test/validated_yamls/templates/obtype_config/atms_n21.yaml:
```diff
@@ -0,0 +1,590 @@
+                                  1.0, 1.0, 1.0, 1.0, 1.0,
+                                  1.0, 1.0, 1.0, 2.0, 4.5,
+                                  4.5, 2.0, 2.0, 2.0, 2.0,
+                                  2.0, 2.0]
```
Lines 534-538: change to:
```
               obserr_bound_max: *atms_n21_obserr_bound_max
```

SamuelDegelia-NOAA replied:
Ah that probably answers my question from above.

guoqing-noaa replied:
Ah you mean `atms_n20.yaml` does not use `&atms_n20_obserr_bound_max` either?  If so, we will need to fix it. Thanks!

xyzemc replied:
"obserr_bound_max" can not be removed from the yaml file, which used in obsfunctions/ObsErrorBoundMW.cc  to do the background check.

guoqing-noaa replied:
Thanks, @xyzemc! We will double-check this and, if needed, make corrections in a new PR for other YAML files.

keenaneure commented:
@guoqing-noaa Made those subsequent changes, and with 'git push' I now return an error I did not see before. Here is the output at the command line:

`To https://github.com/keenaneure/RDASApp
! [rejected]        add_atms_n21 -> add_atms_n21 (fetch first)
error: failed to push some refs to 'https://github.com/keenaneure/RDASApp'
hint: Updates were rejected because the remote contains work that you do not
hint: have locally. This is usually caused by another repository pushing to
hint: the same ref. If you want to integrate the remote changes, use
hint: 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.`

SamuelDegelia-NOAA commented:
> @guoqing-noaa Made those subsequent changes, and with 'git push' I now return an error I did not see before. Here is the output at the command line:
> 
> `To https://github.com/keenaneure/RDASApp ! [rejected] add_atms_n21 -> add_atms_n21 (fetch first) error: failed to push some refs to 'https://github.com/keenaneure/RDASApp' hint: Updates were rejected because the remote contains work that you do not hint: have locally. This is usually caused by another repository pushing to hint: the same ref. If you want to integrate the remote changes, use hint: 'git pull' before pushing again. hint: See the 'Note about fast-forwards' in 'git push --help' for details.`

@keenaneure That is because I synced your branch with the latest develop branch of RDASApp due to a few PRs being merged this morning. You can run these commands to fetch and pull these recent updates to your branch. After that you should be able to push your new commits. 

```
git fetch --all
git pull
```

guoqing-noaa commented:
@keenaneure 
Since you have done `git commit`, if it is only one commit, do this before doing `git pull`:
```
git reset HEAD~1
```

SamuelDegelia-NOAA reviewed (approved):
Everything looks good to me! Thanks again @keenaneure

guoqing-noaa commented on rrfs-test/validated_yamls/templates/obtype_config/atms_n21.yaml:
```diff
@@ -0,0 +1,586 @@
+                                   0.300,  0.300,  0.400,  0.400,  0.400,
+                                   0.450,  0.450,  0.550,  0.800,  4.000,
+                                   4.000,  4.000,  3.500,  3.000,  3.000,
+                                   3.000,  3.000]
```
This error table comes from:
https://github.com/NOAA-EMC/rrfs-workflow/blob/26eca5c018e545582d01d73ad47740dca942b0ea/fix/jedi/satinfo.rrfs.all#L1655-L1676

It appears twice in this YAML file, one is here, one starts from Line 482.

Could you replace both occurrences to:
```
               obserr_clearsky: *&atms_n21_error
```

and add the following anchor name in the anchor section?
```
         _anchor_error: &atms_n21_error
             [ 4.500,  4.500,  4.500,  2.500,  0.550,
               0.300,  0.300,  0.400,  0.400,  0.400,
               0.450,  0.450,  0.550,  0.800,  4.000,
               4.000,  4.000,  3.500,  3.000,  3.000,
               3.000,  3.000]
```
Thanks!

guoqing-noaa commented on rrfs-test/validated_yamls/templates/obtype_config/atms_n21.yaml:
```diff
@@ -0,0 +1,586 @@
+                                   0.300,  0.300,  0.400,  0.400,  0.400,
+                                   0.450,  0.450,  0.550,  0.800,  4.000,
+                                   4.000,  4.000,  3.500,  3.000,  3.000,
+                                   3.000,  3.000]
```
See the comment near line 367. Thanks!

keenaneure commented:
I made the changes, but I am returning an error when I try to git pull:
<img width="569" alt="Screenshot 2025-04-28 at 12 54 48 PM" src="https://github.com/user-attachments/assets/683f3602-6a1c-4b77-a246-a0354a51bb07" />

keenaneure commented:
I cannot commit the changes unless I do git pull, if I understand correctly.

guoqing-noaa commented:
> I cannot commit the changes unless I do git pull, if I understand correctly.

As mentioned, feel free to stop by when you are ready. We will get this sorted out. Thanks!

keenaneure commented:
Pushed these changes now.

guoqing-noaa commented on rrfs-test/validated_yamls/templates/obtype_config/atms_n21.yaml:
```diff
@@ -0,0 +1,584 @@
+             channels: *atms_n21_channels
+             options:
+               channels: *atms_n21_channels
+               obserr_clearsky: *&atms_n21_error
```
YAML linter found this typo.
Change
`obserr_clearsky: *&atms_n21_error`
to
`obserr_clearsky: *atms_n21_error`

guoqing-noaa commented on rrfs-test/validated_yamls/templates/obtype_config/atms_n21.yaml:
```diff
@@ -0,0 +1,584 @@
+               obserr_function:
+                 name: InitialObsError@DerivedMetaData
+                 channels: *atms_n21_channels
+               obserr_clearsky: *&atms_n21_error
```
remove the extra `&` in line 484 as well

xyzemc commented:
> > I see that the new anchor `&atms_n21_obserr_bound_max` is not actually used in the yaml after it is defined. It is the same for `atms_n20.yaml` with `&atms_n20_obserr_bound_max`. But a similar variable is used in `abi_g16.yaml` and `abi_g18.yaml` for a final gross check filter. Is that filter not needed for ATMS data? @xyzemc
> > Thanks @SamuelDegelia-NOAA  to point this out. I will check why atms and abi doing the gross check differently.

@SamuelDegelia-NOAA  After a dig into a little the UFO code, I found that atms and abi use the  obserr_bound_max in ObsErrorBoundIMW and ObsErrorBoundIR obsfunction respectively. So both atms and abi filters require this variable.

keenaneure commented:
Removed the & in both cases and pushed changes just now.

guoqing-noaa commented:
> Removed the & in both cases and pushed changes just now.

@keenaneure , Thanks for addressing all comments! 
Just launched a CI/CD on Hera. Once it passes, it is good to go.
